File: server/paper/src/main/java/su/plo/lib/paper/command/PaperCommandManager.java
package su.plo.lib.paper.command;

import lombok.RequiredArgsConstructor;
import org.bukkit.plugin.java.JavaPlugin;
import org.jetbrains.annotations.NotNull;
import su.plo.lib.api.server.MinecraftServerLib;
import su.plo.lib.api.server.command.MinecraftCommandManager;
import su.plo.lib.paper.chat.BaseComponentTextConverter;

@RequiredArgsConstructor
public final class PaperCommandManager extends MinecraftCommandManager {

    private final MinecraftServerLib minecraftServer;
    private final BaseComponentTextConverter textConverter;

    public synchronized void registerCommands(@NotNull JavaPlugin loader) {
        commandByName.forEach((name, command) -> {
            PaperCommand paperCommand = new PaperCommand(minecraftServer, textConverter, command, name);
            loader.getServer().getCommandMap().register("plasmovoice", paperCommand);

        });
        this.registered = true;
    }
}

File: server/paper/src/main/kotlin/su/plo/lib/paper/command/PaperCommandManager.kt
package su.plo.lib.paper.command

import org.bukkit.command.CommandSender
import org.bukkit.command.SimpleCommandMap
import org.bukkit.entity.Player
import org.bukkit.plugin.java.JavaPlugin
import su.plo.lib.api.server.MinecraftServerLib
import su.plo.lib.api.server.command.MinecraftCommand
import su.plo.lib.api.server.command.MinecraftCommandManager
import su.plo.lib.api.server.command.MinecraftCommandSource
import su.plo.lib.paper.chat.BaseComponentTextConverter

class PaperCommandManager(
    private val minecraftServer: MinecraftServerLib,
    private val textConverter: BaseComponentTextConverter
) : MinecraftCommandManager<MinecraftCommand>() {

    @Synchronized
    fun registerCommands(loader: JavaPlugin) {
        commandByName.forEach { (name, command) ->
            val paperCommand = PaperCommand(this, command, name)

            val commandMap = loader.server.javaClass
                .getDeclaredField("commandMap").also {
                    it.isAccessible = true
                }
                .get(loader.server) as SimpleCommandMap

            commandMap.register("plasmovoice", paperCommand)
        }

        registered = true
    }

    override fun getCommandSource(source: Any): MinecraftCommandSource  {
        require(source is CommandSender) { "source is not ${CommandSender::class.java}" }

        return if (source is Player) minecraftServer.getPlayerByInstance(source)
        else PaperDefaultCommandSource(source, textConverter)
    }
}
